**Даны значения зарплат из выборки выпускников: 100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150. Посчитать
(желательно без использования статистических методов наподобие std, var,
mean):**  

а) среднее арифметическое,  
б) среднее квадратичное отклонение,  
с) смещенную и несмещенную оценки дисперсий для данной выборки.

In [4]:
import numpy as np
from math import factorial

In [5]:
arr=np.array([100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150])
arr

array([100,  80,  75,  77,  89,  33,  45,  25,  65,  17,  30,  24,  57,
        55,  70,  75,  65,  84,  90, 150])

a\) Расчитаем среднее арифметическое по формуле:
$\displaystyle M(X)=\frac{1}{n}\sum^n_{i=1}x_i$

In [6]:
def mean_value(array):
    return sum(array)/len(array)

In [16]:
print(f'Среднее арифметическое для данной выборки М(Х) = {mean_value(arr): .4f}')

Среднее арифметическое для данной выборки М(Х) =  65.3000


Проверим с помощью функции mean:

In [17]:
print(f'Среднее арифметическое для данной выборки М(Х) = {np.mean(arr): .4f}')

Среднее арифметическое для данной выборки М(Х) =  65.30


б) Расчитаем среднее квадратичное отклонение выборки по формуле:
$\sigma=\sqrt{\frac{\displaystyle\sum^n_{i=1}(x_{i}-\bar x)^2}{n}}$

In [9]:
def mean_square_deviation(array):
    square_dev=(array-mean_value(array))**2
    return (sum(square_dev)/len(square_dev))**(1/2)

In [10]:
print(f'Среднее квадратичное отклонение для данной выборки SD = {mean_square_deviation(arr): .4f}')

Среднее квадратичное отклонение для данной выборки SD =  30.8239


Проверим с помощью функции std:

In [18]:
print(f'Среднее квадратичное отклонение для данной выборки SD = {np.std(arr): .4f}')

Среднее квадратичное отклонение для данной выборки SD =  30.8239


в) Расчитаем смещенную оценку дисперсии по формуле:
$\sigma^2=\frac{\displaystyle\sum^n_{i=1}(x_{i}-\bar x)^2}{n}$  
а несмещённую по формуле:
$\sigma^2=\frac{\displaystyle\sum^n_{i=1}(x_{i}-\bar x)^2}{n-1}$

In [12]:
def dispers(array, no_off=False):
#аргумент no_off отвечает за расчет смещенной, или несмещенной оценки дисперсии. По умолчанио расчитывается смещенная.#
    square_dev=(array-mean_value(array))**2
    return sum(square_dev)/(len(square_dev)-1) if no_off else sum(square_dev)/len(square_dev)

In [23]:
print(f'Смещенная оценка дисперсии для данной выборки D = {dispers(arr): .4f}\n'
      f'Неcмещенная оценка дисперсии для данной выборки D = {dispers(arr, True): .4f}')

Смещенная оценка дисперсии для данной выборки D =  950.1100
Неcмещенная оценка дисперсии для данной выборки D =  1000.1158


Проверим с помощью функции var:

In [19]:
print(f'Смещенная оценка дисперсии для данной выборки D = {np.var(arr): .4f}')

Смещенная оценка дисперсии для данной выборки D =  950.1100


In [22]:
print(f'Неcмещенная оценка дисперсии для данной выборки D = {np.var(arr, ddof=1): .4f}')

Неcмещенная оценка дисперсии для данной выборки D =  1000.1158


------------------------------------------------------------------------

#### Ответ:

а) $М(Х)= 65.30$;  
б) $\sigma = 30.8239$;  
в) $\sigma^2_{\text{смещ}}=950.11$,
$\sigma^2_{\text{несмещ}}=1000.1158$.

------------------------------------------------------------------------